roles/vault/defaults/main.yml

```diff
@@ -83,6 +83,8 @@ vault_ca_options:
     format: pem
     ttl: "{{ vault_max_lease_ttl }}"
     exclude_cn_from_sans: true
+    altnames:
+      - "vault.kube-system.svc.cluster.local"
```

Comment:
You should add all the non fully qualified names, too:
vault
vault.kube-system